samples/erc20-sample/chaincode/token_contract.go

```diff
@@ -0,0 +1,734 @@
+
+	// Check if transferred value is less than allowance
+	if currentAllowance < value {
+		return fmt.Errorf("spender does not have enough allowance for transfer")
+	}
+
+	// Initiate the transfer
+	err = transferHelper(ctx, from, to, value)
+	if err != nil {
+		return fmt.Errorf("failed to transfer: %v", err)
+	}
+
+	// Decrease the allowance
+	updatedAllowance, err := sub(currentAllowance, value)
+	if err != nil {
+		return err
+	}
+
+	err = ctx.GetStub().PutState(allowanceKey, []byte(strconv.Itoa(updatedAllowance)))
+	if err != nil {
+		return err
+	}
+
+	// Emit the Transfer event
+	transferEvent := event{from, to, value}
+	transferEventJSON, err := json.Marshal(transferEvent)
+	if err != nil {
+		return fmt.Errorf("failed to obtain JSON encoding: %v", err)
+	}
+	err = ctx.GetStub().SetEvent("Transfer", transferEventJSON)
+	if err != nil {
+		return fmt.Errorf("failed to set event: %v", err)
+	}
+
+	log.Printf("spender %s allowance updated from %d to %d", spender, currentAllowance, updatedAllowance)
+
+	return nil
+}
+
+// Name returns a descriptive name for fungible tokens in this contract
+// returns {String} Returns the name of the token
+
+func (s *SmartContract) Name(ctx contractapi.TransactionContextInterface) (string, error) {
+
+	// Check if contract has been intilized first
+	initialized, err := checkInitialized(ctx)
+	if err != nil {
+		return "", fmt.Errorf("failed to check if contract is already initialized: %v", err)
+	}
+	if !initialized {
+		return "", fmt.Errorf("Contract options need to be set before calling any function, call Initialize() to initialize contract")
+	}
+
+	bytes, err := ctx.GetStub().GetState(nameKey)
+	if err != nil {
+		return "", fmt.Errorf("failed to get Name bytes: %s", err)
+	}
+
+	return string(bytes), nil
+}
+
+// Symbol returns an abbreviated name for fungible tokens in this contract.
+// returns {String} Returns the symbol of the token
+
+func (s *SmartContract) Symbol(ctx contractapi.TransactionContextInterface) (string, error) {
+
+	// Check if contract has been intilized first
+	initialized, err := checkInitialized(ctx)
+	if err != nil {
+		return "", fmt.Errorf("failed to check if contract is already initialized: %v", err)
+	}
+	if !initialized {
+		return "", fmt.Errorf("Contract options need to be set before calling any function, call Initialize() to initialize contract")
+	}
+
+	bytes, err := ctx.GetStub().GetState(symbolKey)
+	if err != nil {
+		return "", fmt.Errorf("failed to get Symbol: %v", err)
+	}
+
+	return string(bytes), nil
+}
+
+// Set information for a token and intialize contract.
+// param {String} name The name of the token
+// param {String} symbol The symbol of the token
+// param {String} decimals The decimals used for the token operations
+func (s *SmartContract) Initialize(ctx contractapi.TransactionContextInterface, name string, symbol string, decimals string) (bool, error) {
```

Comment:
we can define a group of 
- owners(Orgs) who can mint/burn ERC20 tokens
during this initialization.
Means add a new argument `owners []string`